web-app/js/smartR/_angular/directives/runButton.js

```diff
@@ -48,23 +48,28 @@ window.smartRApp.directive('runButton',
                     template_msg.innerHTML = 'Failure: ' + response.statusText;
                 };
 
+                var _finishedRunning =  function() {
+                    template_btn.disabled = false;
+                    processService.onRunning(false);
+                };
+
                 template_btn.onclick = function() {
 
                     scope.storage = {};
                     template_btn.disabled = true;
                     template_msg.innerHTML = 'Creating plot, please wait <span class="blink_me">_</span>';
-                    scope.$emit('on:running', true);
+                    processService.onRunning(true);
 
                     rServeService.startScriptExecution({
                         taskType: scope.script,
                         arguments: scope.arguments
                     }).then(
                         _successCreatePlot,
-                        _failCreatePlot
-                    ).finally(function() {
-                        template_btn.disabled = false;
-                        scope.$emit('on:running', false);
-                    });
+                        _failCreatePlot,
+                        _finishedRunning
```

Comment:
Are you passing the 'finished' function in the wrong place here? I think this should be removed